executor/server.go

```diff
@@ -383,6 +385,15 @@ func (s *Server) createSession(ctx context.Context, etcdCli *clientv3.Client) (m
 	return session, nil
 }
 
+func (s *Server) resetDiscovery(ctx context.Context) (metaStoreSession, error) {
+	session, err := s.discoveryConnector(ctx)
```

Comment:
Now that `discoveryConnector` has replaced the `discovery` object. Why do we have to update discoveryWather instead of using `discovery.Watch` directly?

Reply:
If we use `discovery.Watch` directly, it will create a watch channel each time when that case branch hits.

Reply:
Got it...

Reply:
We'd better to add a comment above `Watch` function to warn the users.